### DISCUSSION TITLE
Please Publish a Detailed Tutorial on Implementation of Google Adsense without losing SSR

### DISCUSSION THREAD

**Discussion by hassanah07** [vercel/next.js#55755] [App Router]
Hello everyone, There is many difficulties when comes to implement google adsense without losing SSR. I have tried many tutorials which is available on internet till date but did not found any solution on that. Even I have taken ChatGPT. I am requesting to everyone please publish a complete tutorial on how we can implement google adsense without losing SSR. Please, If you can Please Publish. And also request to the vercel team that you guys made NextJS to the awesome level where we can do a big task in a simple way by importing your built in packages, But you guys left this things. Please work on it. Thanks

**Comment by shriekdj** [comment]
what type of issue you are facing like unable to add auto ads, manual ad code `ins` or search engine ads. what is the problem you are facing ?

**Comment by shriekdj** [comment]
few days ago, i also had the issue with new app router, i solved it now i hope i can help if needed.

**Comment by hassanah07** [comment]
Please post the solution
